src/middleware/gzip_middleware.py

```
@@ -0,0 +1,387 @@
+                    endpoint=path,
+                    error_type="size_limit",
+                ).inc()
+
+            logger.error(f"Decompression size limit exceeded for {path}: {e}")
+
+            if self.fail_on_error:
+                await self._send_error_response(send, 413, str(e))
+            else:
+                await self.app(scope, receive, send)
+
+        except gzip.BadGzipFile as e:
```

Flake8 (7.3.0)

[error] local variable 'e' is assigned to but never used

(F841)